### DISCUSSION TITLE
HttpClient timeout exception for Whisper API calls

### DISCUSSION THREAD

**Discussion by syntaxerror1024** [RageAgainstThePixel/OpenAI-DotNet#266] [Q&A] [answered]
# Bug Report

## Overview
When trying to make a call to the Whisper API (```Client.AudioEndpoint.CreateTranscriptionAsync```) I constantly get ```HttpClient``` timeout errors for audio files with a duration of over 30 minutes.

This also occasionally happens for shorter duration audio files too.

Exception thrown:
"The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing."

## To Reproduce
1. Create a ```AudioTranscriptionRequest``` and pass in a audio stream greater than 30 minutes while still being less than 25MB in size. I am able to do this by compressing the audio file with FFMpeg before creating ```AudioTranscriptionRequest```. I am also setting the "responseFormat" to ```AudioResponseFormat.Verbose_Json```.
2. Call ``` await Client.AudioEndpoint.CreateTranscriptionAsync(request)```
3. Timeout exception is thrown.

## Expected behavior
Allow the HTTP request to finish or allow the Timeout in seconds to be configurable. Open AI is clearly taking longer than 100 seconds to create the transcription I am submitting.

## Screenshots
N/A

## Additional context
I am working with a lot of long duration audio files. I cannot split the files when submitting to the Whisper API, because the Verbose JSON response will have the incorrect start and end times (because Whisper is treating them as separate audio files). My only option is to compress the audio files and submit them to Whisper.

**Comment by StephenHodgson** [comment, marked as answer]
Can you [configure the HttpClient](https://github.com/RageAgainstThePixel/OpenAI-DotNet#handling-openaiclient-and-httpclient-lifecycle) timeout by passing it in the constructor of the OpenAIClient:
```csharp
using var customHttpClient = new HttpClient();
// set custom http client properties here
customHttpClient.Timeout = TimeSpan.FromMinutes(30);
var api = new OpenAIClient(client: customHttpClient);
```

**Reply by syntaxerror1024** [reply]
Thank you sir. I did not realize that I can pass a custom HttpClient to the OpenAIClient!
